File: Modern/src/main/scala/prv/zielony/transaction/analyst/modern/services/AccountsService.scala
package prv.zielony.transaction.analyst.modern.services

import org.apache.spark.rdd.RDD
import prv.zielony.transaction.analyst.modern.dao
import prv.zielony.transaction.analyst.modern.model.UserAccount

import prv.zielony.transaction.analyst.modern.dao._

import scala.concurrent.Future

/**
 * Created by zielony on 28.02.16.
 */
object AccountsService {

  def countAllAccountsByCountry(country:String):Future[Long] = {
    dao.userAccounts.countAll map(_.getOrElse(0))
  }

  def calculateAverageBalanceByCountry(country:String):Future[Double] = {

    val futureBalances = userAccounts.findBalanceByCountry(country)

    futureBalances map { balances =>

      balances.map { balance =>
        balance / balances.length
      } reduce { (first, second) =>
        first + second
      }
    }
  }
}

File: Classical/src/main/java/prv/zielony/transaction/analyst/classical/services/AccountsService.java
package prv.zielony.transaction.analyst.classical.services;

import prv.zielony.transaction.analyst.classical.model.Account;

/**
 * Created by zielony on 27.02.16.
 */
public interface AccountsService {
    double getAverageBalanceByCountry(String country, Integer minAge, Integer maxAge);

    int getAccountNumberForCountry(String country);

    Account findAccount(Long id);

    Account createAccount(Long ownerId);

    void removeAccount(Long accountId);
}
